examples/onnx-inference/README.md

```diff
@@ -1,4 +1,104 @@
 # ONNX Inference
 
-This crate provides a simple example for importing ONNX model to Burn.
+This crate provides a simple example for importing MNIST ONNX model to Burn. The onnx file is
+converted into a Rust source file using `burn-import` and the weights are stored in and loaded from
+a binary file.
 
+## Usage
+
+```bash
+cargo run -- 15
+```
+
+Output:
+
+```bash
+Finished dev [unoptimized + debuginfo] target(s) in 0.13s
+    Running `burn/target/debug/onnx-inference 15`
+
+Image index: 15
+Success!
+Predicted: 5
+Actual: 5
+See the image online, click the link below:
+https://datasets-server.huggingface.co/assets/mnist/--/mnist/test/15/image/image.jpg
+```
+
+## How to import
+
+1. Create `model` directory under `src`
+2. Copy the ONNX model to `src/model/mnist.onnx`
+3. Add the following to `mod.rs`:
+   ```rust
+   pub mod mnist {
+       include!(concat!(env!("OUT_DIR"), "/model/mnist.rs"));
+   }
+   ```
+4. Add the modul to lib.rs:
+
+   ```rust
+   pub mod model;
+   ```
+
+5. Add the following to `build.rs`:
+
+   ```rust
+   use burn_import::onnx::ModelGen;
+
+   fn main() {
+       // Generate the model code from the ONNX file.
+       ModelGen::new()
+           .input("src/model/mnist.onnx")
+           .out_dir("model/")
+           .run_from_script();
+   }
+
+   ```
+
+6. Run `cargo build` to generate the model code and weights.
+
+7. Use the model in your code (Sample code):
+
+   ```rust
+   use burn::tensor;
+   use burn_ndarray::NdArrayBackend;
+   use onnx_inference::model::mnist::{Model, INPUT1_SHAPE};
+
+   fn main() {
+       // Create a new model and load the state
+       let model: Model<Backend> = Model::new().load_state();
```

Comment:
At some point I think we should be able to pass the state path as parameter to `load_state`.

Reply:
Yeah, I agree we should more options. I'll file a ticket to enhance and gather requirements. Please see my PR comment which was replied to your top comment: https://github.com/burn-rs/burn/pull/319#issuecomment-1529074801

Reply:
Here is the ticket to enhance state loading: https://github.com/burn-rs/burn/issues/321